test/websocket-test.js

```diff
@@ -0,0 +1,167 @@
+/* eslint-env mocha */
+/* eslint prefer-arrow-callback: "off" */
+
+/**
+ * Tests for websockets - test/websocket-test.js
```

Comment:
There are some websocket tests in `http-test` already (see #739). I think this is ok though - I'm a fan of more files for the testing suite rather than try to get all related tests in the same files. [Bitcoin Core has loads of individual test files](https://github.com/bitcoin/bitcoin/tree/master/test/functional)

Reply:
I saw that and didn't want to add these tests to that file since this file needs to simulate reorgs which would require adding a lot to that file. Open to moving the websocket tests in `http-test` to this file.